### PR TITLE
Add check config for endpoints matcher

### PR DESCRIPTION
**Pull request by desurd**
associated issue: #3044

## Contributor Checklist:

* [ ] I have updated the unit tests

### CODE REVIEW

**Comment by desurd** [comment]
I have some doubts about the creation of new unit tests. So i did not add new ones. 
However, feel free to add any comments and where the units tests should be added in this case

**Comment by noc0lour** [comment]
for reference: #3044

File: master/buildbot/www/authz/endpointmatchers.py [hunk 1]
```diff
@@ -32,6 +33,13 @@ def __init__(self, role, defaultDeny=True):
         self.defaultDeny = defaultDeny
         self.owner = None
 
+    def checkConfig(self, **kwargs):
+        '''
+        goal: check the configuration about the EndpointMatcherBase
+        '''
+        if 'role' is not in kwargs:
+            config.error("role needs to present in the configuration file")
+
```
Comment (tardyp) [inline master/buildbot/www/authz/endpointmatchers.py]: need to call base class

File: master/buildbot/www/authz/endpointmatchers.py [hunk 2]
```diff
@@ -110,6 +118,11 @@ class AnyControlEndpointMatcher(EndpointMatcherBase):
     def __init__(self, **kwargs):
         EndpointMatcherBase.__init__(self, **kwargs)
 
+    def checkConfig(self, **kwargs):
+        if 'role' in kwargs:
+            if type(kwargs[role]) is not 'string':
+                config.error("role type needs to be a string")
+
```
Comment (tardyp) [inline master/buildbot/www/authz/endpointmatchers.py]: idem

**Pull request closed without merging**